credentials/oauth/oauth.go

```diff
@@ -129,11 +129,26 @@ func NewComputeEngine() credentials.Credentials {
 
 // serviceAccount represents credentials via JWT signing key.
 type serviceAccount struct {
-	config *jwt.Config
-}
-
-func (s serviceAccount) GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error) {
-	token, err := s.config.TokenSource(ctx).Token()
+	config      *jwt.Config
+	tokenSource oauth2.TokenSource
+}
+
+func (s *serviceAccount) GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error) {
+	var token *oauth2.Token
+	var err error
+	if s.tokenSource != nil {
+		token, err = s.tokenSource.Token()
+		if err != nil {
```

Comment:
This if block should be redundant. Token() shouldn't return a non-nil token with a non-nil error.